lib/Backend/Commands.hs

```diff
@@ -482,3 +478,45 @@ getEntryOrDir path =
                       \ entry or directory name."
                     , "Got: '" <> secret <> "'."
                     ]
+
+-- | This function gets all entries, that are exist in given entry path.
+--
+-- Note: the root path @/@ cannot possibly be occupied by an entry,
+-- therefore we skip the check for that path.
+getEntriesInEntryPath :: forall r. Member BackendEffect r => EntryPath -> Sem r [EntryPath]
+getEntriesInEntryPath entryPath = do
+  let parentDirsExceptRoot = entryPath
+        & Path.entryPathParentDirs
+        & NE.toList
+        -- Ignore paths that cannot possibly point to an entry
+        <&> Path.pathAsEntryPath
+        & rights
+
+  filterM pathIsEntry parentDirsExceptRoot
+
+checkCreateEntry
+  :: forall r
+   . (Member BackendEffect r)
+  => Bool -> Entry -> Sem r CreateCheckResult
```

Comment:
I think returning a `Validation [CreateError] Entry` would be semantically better.

If you're not familiar with it: [`Validation e a`](https://hackage.haskell.org/package/either-5.0.1.1/docs/Data-Either-Validation.html#t:Validation) is very similar to `Either e a`, except:
1. `Validation` has an applicative instance but no monad instance.
2. When you `sequence` many `Either`s, you only get the first error and all the other items are ignored. On the other hand, when you `sequenceA` many `Validation`s, all the errors all collated together (which is the behaviour we want here).

After changing `checkCreateEntry` to return `Sem r (Validation [CreateError] Entry)`, you can collate all the errors using `for + sequenceA`:

```hs
do
  let entries = [] :: [Entry]
  let force = True
 
  sequenceA <$> for entries (checkCreateEntry2 force) >>= \case
    Failure allErrors -> pure ()
    Success allEntries -> pure ()
```

Reply:
Done